SimG4Core/Geometry/interface/SensitiveDetectorCatalog.h

```diff
@@ -3,20 +3,23 @@
 
 #include <map>
 #include <string>
+#include <string_view>
 #include <vector>
+#include <unordered_set>
 
 class SensitiveDetectorCatalog {
 public:
-  using MapType = std::map<std::string, std::vector<std::string>>;
+  using MapType = std::map<std::string, std::unordered_set<std::string>>;
   void insert(const std::string &, const std::string &, const std::string &);
-  const std::vector<std::string> &logicalNames(const std::string &readoutName) const;
-  const std::vector<std::string> &readoutNames(const std::string &className) const;
-  std::vector<std::string> readoutNames() const;
-  std::string className(const std::string &readoutName) const;
+  const std::vector<std::string_view> logicalNames(const std::string &readoutName) const;
```

Comment:
@ianna could you please clarify the advantage of moving from ```std::string``` to ```std::string_view```?

Reply:
@fabiocos  - ```std::string_view``` is used to avoid an ```std::string``` copy. https://en.cppreference.com/w/cpp/string/basic_string_view

Reply:
@ianna , @fabiocos , I agree that the order of SD is not important. Concerning std::string and std::string_view I am not 100% sure if we save anything at this point. In any case, strings are kept in the Catalog and we have access to string via reference. I personally use in my code references as much as I can, avoiding copies of string. If std::string_view allows doing this with compiler control - fine, let us use it.

Reply:
@civanch - FYI, igprof reports 0.1% improvement running ZMM_13TeV_TuneCUETP8M1_cfi_GEN_SIM step1 on 10 events.